_posts/2023-04-19-12.0.0-release.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+title: "Apache Arrow 12.0.0 Release"
+date: "2023-04-19 00:00:00"
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+
+The Apache Arrow team is pleased to announce the 12.0.0 release. This covers
+over 3 months of development work and includes [**XXX resolved issues**][1]
+from [**YYY distinct contributors**][2]. See the [Install Page](https://arrow.apache.org/install/)
+to learn how to get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+Since the 11.0.0 release, Wang Mingming, Mustafa Akur and Ruihang Xia
+have been invited to be committers.
+Will Jones have joined the Project Management Committee (PMC).
+
+Thanks for your contributions and participation in the project!
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+
+## C++ notes
+
+## C# notes
+
+## Go notes
+
+## Java notes
+
+## JavaScript notes
+
+## Python notes
```

Comment:
```suggestion
## Python notes

Compatibility notes:

* As Plasma has been removed in this release [GH-33243](https://github.com/apache/arrow/issues/33243) the serialization module in PyArrow was also removed [GH-29705](https://github.com/apache/arrow/issues/29705). IPC (Inter-Process Communication) functionality of pyarrow or the standard library pickle should be used instead.
* Deprecated `use_async` has now been removed from the dataset module [GH-30774](https://github.com/apache/arrow/issues/30774)
* Minimum Cython version to build PyArrow from source has been raised to 0.29.31 [GH-34933](https://github.com/apache/arrow/issues/34933)

New features:

* FixedShapeTensorType binding has been implemented [GH-34882](https://github.com/apache/arrow/issues/34882) and [GH-34956](https://github.com/apache/arrow/issues/34956)
* Run-End Encoded arrays binding has been implemented [GH-34686](https://github.com/apache/arrow/issues/34686) together with RunEndEncodedScalar values binding [GH-34568](https://github.com/apache/arrow/issues/34568)
* Method `is_nan` has been added to Array, ChunkedArray and Expression [GH-34154](https://github.com/apache/arrow/issues/34154)
* `pyarrow.acero` module initial binding has been implemented [GH-33976](https://github.com/apache/arrow/issues/33976)
* Dataframe interchange protocol has been implemented for RecordBatch [GH-33926](https://github.com/apache/arrow/issues/33926)

Other improvements:

* Extension arrays can now be concatenated [GH-31868](https://github.com/apache/arrow/issues/31868)
* `get_partition_keys` helper function is implemented in the `dataset` module to access the partitioning field's key/value from the partition expression of a certain dataset fragment [GH-33825](https://github.com/apache/arrow/issues/33825)
* PyArrow can now be compiled using Cython 3 [GH-34564](https://github.com/apache/arrow/issues/34564)
* [Arrow Archery tool](https://arrow.apache.org/docs/dev/developers/continuous_integration/archery.html) now includes linting of the Cython files [GH-31905](https://github.com/apache/arrow/issues/31905)
* Array objects can now be accepted by the `pa.array()` constructor [GH-34411](https://github.com/apache/arrow/issues/34411)
*  The default row group size when writing parquet files has been changed [GH-34280](https://github.com/apache/arrow/issues/34280)
* RecordBatch has the `select()` method implemented [GH-34359](https://github.com/apache/arrow/issues/34359)
* New method `drop_column` on the `pyarrow.Table` supports passing a single column as a string [GH-33377](https://github.com/apache/arrow/issues/33377)
* User-defined tabular functions, which are a user-functions implemented in Python that return a stateful stream of tabular data, are now also supported [GH-32916](https://github.com/apache/arrow/issues/32916) 


Relevant bug fixes:

* Acero can now detect and raise an error in case a join operation needs too much bytes of key data [GH-34474](https://github.com/apache/arrow/issues/34474)
* Fix for converting non-sequence object in `pa.array()` [GH-34944](https://github.com/apache/arrow/issues/34944)
* Fix erroneous table conversion to pandas if table includes an extension array that does not implement `to_pandas_dtype` [GH-34906](https://github.com/apache/arrow/issues/34906)
* Reading from a closed `ArrayStreamBatchReader` now returns invalid status instead of segfaulting https://github.com/apache/arrow/issues/34165
* `array()` now returns `pyarrow.Array` and not `pyarrow.ChunkedArray` for columns with `__arrow_array__` method and only one chunk so that the conversion of pandas dataframe with categorical column of dtype `string[pyarrow]` does not fail [GH-33727]https://github.com/apache/arrow/issues/33727)
* Custom type mapper in `to_pandas` now converts index dtypes together with column dtypes [GH-34283](https://github.com/apache/arrow/issues/34283)
```